src/main/mondrian/rolap/RolapNativeSet.java

```diff
@@ -260,6 +265,32 @@ protected TupleList executeList(final SqlTupleReader tr) {
                         dataSource, partialResult, newPartialResult);
             }
 
+            // Did not get as many members as expected - try to complete using
+            // less constraints
+            if (completeWithNullValues && result.size() < maxRows) {
+                RolapLevel l = args[0].getLevel();
+                List<RolapMember> list = new ArrayList<RolapMember>();
+                for (List<Member> lm : result) {
+                    for (Member m : lm) {
+                        list.add((RolapMember) m);
+                    }
+                }
+                SqlTupleReader str = new SqlTupleReader(
+                    new MemberExcludeConstraint(
+                        list, l,
+                        constraint instanceof SetConstraint
+                        ? (SetConstraint) constraint : null));
+                str.setAllowHints(false);
```

Comment:
Sure, the value is set, but it's not used anywhere.  What did you intend to do with it?